### ISSUE TITLE
UI artefact Worldclock

### ISSUE THREAD

**Issue by stefonarch** [lxqt/lxqt#1734]
Under some circumstances the plugin has both focus and normal background color, see picture.

![schermata-24-11-55-36](https://user-images.githubusercontent.com/10681413/61784758-8e311c80-ae0a-11e9-8286-c304b0d3dd2e.png)

It get's normal again when hovering the cursor on it.

##### Expected Behavior

Color is always consistent
##### Current Behavior

Sometimes clock is seen like in the  picture
##### Possible Solution

Possible bug that plugin remain in  focus when not closing explicitly calendar?
##### Steps to Reproduce (for bugs)

1. Open calendar, click on some other window to focus it, calendar closes but background color  of the world clock is still light grey (focus)  on my theme.
2. It's hard to find what exactly triggers the phenomenon, it can be  start typing in a text field, switching between windows, at some point
3. Text background get's dark (loses focus color), other background not.

                                                       --->
* Distribution & Version: up to date arch and git checkout

**Comment by tsujan** [comment]
Most probably, it's Qt's infamous "hover bug" (as I call it). It shows up in various forms with Qt5 and has never been fixed (on some occasions, it has been denied). To reproduce it:

(1) Right click on a widget that (a) has a right-click menu and (b) whose hover state is shown by the style (or stylesheet). It can be any widget inside any app.

(2) When the right-click menu is opened, close it by clicking somewhere else (on desktop, for example).

Then, the hover state will persist until you move the cursor over the widget.

You could also see the same bug in scrollbars, provided that the widget style you use shows their hover state (that's the case with Breeze and Kvantum). Nothing can be done in the app; it should be fixed in Qt.

**Comment by stefonarch** [comment]
Yeah, looks  like this. It's only in the worldclock so bicolor because there are 2 different style containers or how you call them.

```
 * WorldClock
 */
#WorldClock:hover {
   border: 1px solid rgba(0, 0, 0, 50%);
   background: rgba(255, 255, 255, 20%);
}

#WorldClockContent {
    margin-right: 2px;
    margin-left: 2px;
    padding: 2px;
    color: #DEDEDE;
    background: transparent;
}

```

**Comment by tsujan** [comment]
I've removed most hover states from my LXQt panel theme to get rid of it; only the task-bar and quick launch have the hover state here -- I'd seen the "hover bug" even in systray (notification area)!

**Comment by Hadi374** [comment]
quick launch don't have "hover bug" in my system. when i right click on an item in quick launch right click menu appears and at same time color of hovered item resets to defaut.
this is good to reset color when we right click and right click menu appears .

**Comment by tsujan** [comment]
> quick launch don't have "hover bug" in my system.

With X11 + Qt5, the hover bug is everywhere ;) It isn't just what you see (some Qt devs may have made the same mistake with Fusion).

However, there are ways to *hide* it in Panel themes, even with a hover background. Luckily, some themes don't show it but I can easily make a theme that shows it.

Perhaps, as a workaround, we should edit all of our themes, so that it isn't shown with any of them. In that case, I need a list of those LXQt themes that can't hide it. @stefonarch ?

**Comment by stefonarch** [comment]
> Perhaps, as a workaround, we should edit all of our themes, so that it isn't shown with any of them. In that case, I need a list of those LXQt themes that can't hide it. @stefonarch ?

dark and ambiance as far as I can say.

**Comment by tsujan** [comment]
@stefonarch Do you use a mouse gesture program (easystroke or KWin's mouse gestures)?

**Comment by stefonarch** [comment]
No.

**Comment by tsujan** [comment]
> No.

This Qt5 mouse-over problem is very confusing. It's especially annoying with a mouse gesture program; without it, it disappears in some places (@Hadi374 mentioned quick launch) but persists in others (task-bar and world-clock, for example). Qt4 didn't have it, as GTK3 doesn't. Something is still wrong in Qt's handling of X11.

**Comment by tsujan** [comment]
We can't do anything about this other than removing all hover blocks from LXQt themes, which isn't desirable.

Let's hope Qt6 has fixed it. There are fundamental changes in the code of Qt6's mouse events.

**Comment by stefonarch** [comment]
I must say that using  the Valendas theme I didn't see those anymore, probably because there's no hover there....

**Comment by tsujan** [comment]
Here I see them in Quick Launch once in a while.
